deploy/Chart/templates/controller/webhook.yaml

```diff
@@ -0,0 +1,86 @@
+{{- $existingSecret := lookup "v1" "Secret" .Release.Namespace "controller-cert"}}
+{{- $existingWebhook := lookup "admissionregistration.k8s.io/v1" "ValidatingWebhookConfiguration" .Release.Namespace "recipe-webhook.radapp.io"}}
+{{- $ca := genCA "controller-ca" 3650 }}
+{{- $cn := printf "controller" }}
+{{- $altName1 := printf "controller.%s" .Release.Namespace }}
+{{- $altName2 := printf "controller.%s.svc" .Release.Namespace }}
+{{- $altName3 := printf "controller.%s.svc.cluster" .Release.Namespace }}
+{{- $altName4 := printf "controller.%s.svc.cluster.local" .Release.Namespace }}
+{{- $cert := genSignedCert $cn nil (list $altName1 $altName2 $altName3 $altName4) 3650 $ca }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: controller-cert
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: controller
+    app.kubernetes.io/part-of: radius
+data:
+  {{ if $existingSecret }}tls.crt: {{ index $existingSecret.data "tls.crt" }}
+  {{ else }}tls.crt: {{ b64enc $cert.Cert }}
+  {{ end }}
+
+  {{ if $existingSecret }}tls.key: {{ index $existingSecret.data "tls.key" }}
+  {{ else }}tls.key: {{ b64enc $cert.Key }}
+  {{ end }}
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: recipe-webhook.radapp.io
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    caBundle: {{ b64enc $ca.Cert }}
+    service:
+      name: controller
+      namespace: {{ .Release.Namespace }}
+      path: /validate-radapp-io-v1alpha3-recipe
+  failurePolicy: Fail
+  matchPolicy: Equivalent
+  name: recipe-webhook.radapp.io
+  rules:
+  - apiGroups:
+    - radapp.io
+    apiVersions:
+    - v1alpha3
+    operations:
+    - CREATE
+    - UPDATE
+    resources:
+    - recipes
+  sideEffects: None
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: deployment-webhook.radapp.io
+  objectSelector:
+    matchLabels:
+      radapp.io/enabled: "true"
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    caBundle: {{ b64enc $ca.Cert }}
+    service:
+      name: controller
+      namespace: {{ .Release.Namespace }}
+      path: /mutate-radapp-io-v1alpha3-builtindeployment
+  failurePolicy: Fail
+  matchPolicy: Equivalent
+  name: deployment-webhook.radapp.io
+  rules:
+  - apiGroups:
+    - radapp.io
+    apiVersions:
+    - v1alpha3
+    operations:
+    - CREATE
+    - UPDATE
+    resources:
+    - deployments
```

Comment:
I think a wire got crossed here, I was expecting this to handle the built-in Kubernetes `Deployment` resource type: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

We don't need to define our own CRD, we want to provide a webhook for a type that's already defined by Kubernetes.

Reply:
oops! i updated the apiGroups and apiVersions to ensure it is built in deployment type.